### PR TITLE
chore: Migrations for #10946

### LINKED ISSUE #10946: Clean up constructors after #10945
https://github.com/SAP/spartacus/issues/10945 added dependencies to existing components. Let's drop the deprecated versions in next Major and make the previously-optional dependencies required in classes:
- ProductListItemComponent
- ProductGridItemComponent
- CartItemComponent

Some new components were added in https://github.com/SAP/spartacus/issues/10945, but they could not claim a new class `CartItemContext` as required (because customers may have not provided it in their custom cart item component). But in Major we cas make `CartItemContext` a required dependency, in components:
- ConfiguratorIssuesNotificationComponent
- ConfiguratorCartEntryInfoComponent

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-10946

### CODE REVIEW

File: feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts [hunk 1]
```diff
@@ -1,47 +1,26 @@
...
-   */
-  constructor(cartItemContext: CartItemContext);
-
   constructor(
-    // TODO(#10946): make CartItemContext a required dependency and drop fallbacks to `?? EMPTY`.
-    @Optional() protected cartItemContext?: CartItemContext,
-    @Optional()
-    protected commonConfigUtilsService?: CommonConfiguratorUtilsService
+    protected cartItemContext: CartItemContext,
+    protected commonConfigUtilsService: CommonConfiguratorUtilsService
```
Comment (bgambocjaviniar) [inline feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts:15]: Can the reviewer double check if it is really fine? 
In line 13 is says to make it required, but we're using it as optional in line 47

line 13 and line 47 from the old changes.

Comment (WeizhengSap) [reply feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts:15]: I think it is fine. But in line 47, it should be `readonly shouldShowButton$: Observable<boolean> = this.commonConfigUtilsService.isActiveCartContext(this.cartItemContext);`

Comment (WeizhengSap) [reply feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts:15]: also need to remove `?` from `this.commonConfigUtilsService?` in this class.

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-issues-notification.component.migration.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const CONFIGURATOR_ISSUES_NOTIFICATION_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  // feature-libs/product-configurator/common/components/configurator-issues-notification/configurator-issues-notification.component.ts
+  class: CONFIGURATOR_ISSUES_NOTIFICATION_COMPONENT,
+  importPath: SPARTACUS_PRODUCT_CONFIGURATOR_COMMON,
+  deprecatedParams: [
+    {
+      className: COMMON_CONFIGURATOR_UTILS_SERVICE,
```
Comment (WeizhengSap) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-issues-notification.component.migration.ts:16]: deprecatedParam is `CartItemContext`, and addParam is `CommonConfiguratorUtilsService`, right?

**Review by WeizhengSap — CHANGES_REQUESTED** [review]
see my comments

**Review by WeizhengSap — APPROVED** [review]

**Comment by marlass** [comment]
Merged to `epic/4-0-deprecations`

**Pull request closed without merging**